backend/api/quivr_api/modules/dependencies.py

```diff
@@ -56,24 +79,93 @@ def get_sync_session() -> Generator[Session, None, None]:
+#     embeddings = get_embedding_client()
+#     supabase_client: Client = get_supabase_client()
+#     documents_vector_store = CustomSupabaseVectorStore(  # Modified by @chloe Check
+#         supabase_client, embeddings, table_name="vectors", vector_service=vector_service
+#     )
+#     return documents_vector_store
+
+
 async def get_async_session() -> AsyncGenerator[AsyncSession, None]:
     async with AsyncSession(
         async_engine, expire_on_commit=False, autoflush=False
     ) as session:
         yield session
 
 
-def get_repository(repository_model: Type[R]) -> Callable[..., R]:
+def get_repository(repository_model: Type[R], asynchronous=True) -> Callable[..., R]:
     def _get_repository(session: AsyncSession = Depends(get_async_session)) -> R:
         return repository_model(session)
 
-    return _get_repository
+    def _get_sync_repository(session: Session = Depends(get_sync_session)) -> R:
+        return repository_model(session)
+
+    if asynchronous:
+        return _get_repository
+    return _get_sync_repository
+
+
+def get_embedding_client() -> Embeddings:
+    global _embedding_service
+    if settings.ollama_api_base_url:
+        embeddings = OllamaEmbeddings(
+            base_url=settings.ollama_api_base_url,
+        )  # pyright: ignore reportPrivateUsage=none
+    else:
+        embeddings = OpenAIEmbeddings()  # pyright: ignore reportPrivateUsage=none
+    return embeddings
+
+
+def get_pg_database_engine():
+    global _db_engine
+    if _db_engine is None:
+        logger.info("Creating Postgres DB engine")
+        _db_engine = create_engine(settings.pg_database_url, pool_pre_ping=True)
+    return _db_engine
+
+
+def get_pg_database_async_engine():
+    global _db_engine
+    if _db_engine is None:
+        logger.info("Creating Postgres DB engine")
+        _db_engine = create_engine(settings.pg_database_async_url, pool_pre_ping=True)
+    return _db_engine
+
+
+async def get_supabase_async_client() -> AsyncClient:
+    global _supabase_async_client
+    if _supabase_async_client is None:
+        logger.info("Creating Supabase client")
+        _supabase_async_client = await create_async_client(
+            settings.supabase_url, settings.supabase_service_key
+        )
+    return _supabase_async_client
+
+
+def get_supabase_client() -> Client:
+    global _supabase_client
+    if _supabase_client is None:
+        logger.info("Creating Supabase client")
+        _supabase_client = create_client(
+            settings.supabase_url, settings.supabase_service_key
+        )
+    return _supabase_client
+
+
+def get_supabase_db() -> SupabaseDB:
+    global _supabase_db
+    if _supabase_db is None:
+        logger.info("Creating Supabase DB")
+        _supabase_db = SupabaseDB(get_supabase_client())
+    return _supabase_db
 
 
-def get_service(service: Type[S]) -> Callable[..., S]:
+def get_service(service: Type[S], asynchronous=True) -> Callable[..., S]:
```

Comment:
add type hint ... This file should always type check

Reply:
it's already type hinted no ?

Reply:
@AmineDiro